### ISSUE TITLE
Crash on VPN billing

### ISSUE THREAD

**Issue by SergeyZhukovsky** [brave/brave-browser#27751]
I found that crash stack on GPS on 1.46.x stable build. It looks like we need to limit retries in that place https://github.com/brave/brave-core/blob/1.46.x/android/java/org/chromium/chrome/browser/vpn/utils/InAppPurchaseWrapper.java#L181

Crash stack:
```
Exception java.lang.IllegalStateException: Too many bind requests(999+) for service Intent { act=com.android.vending.billing.InAppBillingService.BIND pkg=com.android.vending cmp=com.android.vending/com.google.android.finsky.billing.iab.InAppBillingService (has extras) }
  at android.app.ContextImpl.bindServiceCommon (ContextImpl.java:2115)
  at android.app.ContextImpl.bindService (ContextImpl.java:2024)
  at android.content.ContextWrapper.bindService (ContextWrapper.java:870)
  at com.android.billingclient.api.BillingClientImpl.startConnection (BillingClientImpl.java:55)
  at org.chromium.chrome.browser.vpn.utils.InAppPurchaseWrapper.connectToBillingService (InAppPurchaseWrapper.java:93)
  at org.chromium.chrome.browser.vpn.utils.InAppPurchaseWrapper$1.onBillingServiceDisconnected (InAppPurchaseWrapper.java:202)
  at com.android.billingclient.api.zzaf.onServiceDisconnected (com.android.billingclient:billing@@4.0.0:4)
  at android.app.LoadedApk$ServiceDispatcher.doConnected (LoadedApk.java:2341)
  at android.app.LoadedApk$ServiceDispatcher$RunConnection.run (LoadedApk.java:2381)
  at android.os.Handler.handleCallback (Handler.java:942)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loopOnce (Looper.java:226)
  at android.os.Looper.loop (Looper.java:313)
  at android.app.ActivityThread.main (ActivityThread.java:8741)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:571)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1067)
```

**Comment by kjozwiak** [comment]
The above requires 1.47.181 or higher for 1.47.x verification 👍 @Uni-verse @srirambv STR/Cases can be found via https://github.com/brave/brave-core/pull/16795#issuecomment-1401416857. Basically spot checking VPN and ensuring everything is working as expected as we don't have exact STR/Cases as the crash report came form the GPS dashboard.

**Comment by Uni-verse** [comment]
Verified on `Samsung GS 21` using version: 

```
Brave	1.47.182 Chromium: 109.0.5414.87 (Official Build) (64-bit) 
Revision	2dc18eb511c56e012081b4abc9e38c81c885f7d4-refs/branch-heads/5414@{#1241}
OS	Android 13; Build/TP1A.220624.014
```

- Verified completing purchase flow and VPN functionality

Example | Example | Example | Example | Example | Example | Example
-|-|-|-|-|-|-
![screenshot-1674572338156](https://user-images.githubusercontent.com/17885425/214330265-5d315cd1-0a0a-48f6-8668-99968af83736.png) | ![screenshot-1674572345285](https://user-images.githubusercontent.com/17885425/214330270-49e67dd6-3394-4d1e-95e7-569977bc2188.png) | ![screenshot-1674572358746](https://user-images.githubusercontent.com/17885425/214330276-c8187d32-d49e-4ec4-a3be-adda4b0f1464.png) | ![screenshot-1674572366576](https://user-images.githubusercontent.com/17885425/214330279-a4802b27-b52b-472d-b729-4e281b210fb7.png) | ![screenshot-1674572376716](https://user-images.githubusercontent.com/17885425/214331273-2c7d23b2-6458-45fb-b44e-542231c4d408.png) | ![screenshot-1674572426132](https://user-images.githubusercontent.com/17885425/214331277-1fad8fc2-4701-4e20-b3cf-488fc670876a.png) | ![screenshot-1674572604788](https://user-images.githubusercontent.com/17885425/214331279-ad9dc4e7-cb7a-4b9b-9ffa-74d4b9c7976a.png)

**Comment by SergeyZhukovsky** [comment]
That is still a problem and I see it in 1.47.x in GPS. We need to take a look on what could cause it as it looks like it's not related to retries.

**Comment by kjozwiak** [comment]
@SergeyZhukovsky I moved the above back into `1.48.x - Release #4` as the original PR associated with this fix was merged/released via `1.48.x - Release #4` and mentioned in the release notes. I created a duplicate via https://github.com/brave/brave-browser/issues/28946. I also updated both https://github.com/brave/brave-core/pull/17495 and https://github.com/brave/brave-core/pull/17499 to use https://github.com/brave/brave-browser/issues/28946.

**Comment by SergeyZhukovsky** [comment]
@kjozwiak sounds good, I will duplicate such issues in the future.
